### PR TITLE
[BottomSheetDialogFragment] Added constructor that takes a layout ID

### LINKED ISSUE #1105: [BottomSheetDialogFragment] LayoutId in constructor
Since https://issuetracker.google.com/issues/150327080 it is possible to use LayoutId in DialogFragments, it would be nice to be able to do the same with BottomSheetDialogFragment.

### PR DESCRIPTION
**Pull request by 4brunu**
#### Don't forget:

- [X] Identify the component the PR relates to in brackets in the title.
  `[Buttons] Updated documentation`
- [X] Link to GitHub issues it solves. `closes #1105`
- [X] Sign the CLA bot. You can do this once the pull request is opened.

[Contributing](https://github.com/material-components/material-components-android/blob/master/docs/contributing.md)
has more information and tips for a great pull request.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ ext {
 
   androidXVersions = [
     annotation            : '1.0.1',
-    appCompat             : '1.1.0',
+    appCompat             : '1.4.0-alpha02', // Need AppCompatDialogFragment constructor that takes a layout ID
```
Comment (alishari) [inline build.gradle:28]: We should not use the alpha version of a library in this project.

Comment (4brunu) [reply build.gradle:28]: The functionality needed for this PR will only available in AppCompat 1.4.0, so we will need to way for the stable version of it.

**Review by akkapon75 — APPROVED** [review]

**Review by akkapon75 — APPROVED** [review]

**Review by akkapon75 — APPROVED** [review]

**Comment by 4brunu** [comment]
@drchen could you please help to review and merge this PR? Thanks 🙂

**Review by drchen — APPROVED** [review]

**Comment by drchen** [comment]
Hi can you merge the master branch again? There are conflicts when I tried to copy the change.

**Comment by 4brunu** [comment]
I just noticed that in order to update AppCompat to version 1.4.0, it requires also to update the java version to 1.8, so I added one more commit.
https://developer.android.com/jetpack/androidx/releases/appcompat#1.4.0

**Comment by 4brunu** [comment]
I tried to merge the master branch again, but it showed no changes :/

**Comment by 4brunu** [comment]
@drchen any news on this? 🙂

**Comment by josefigueroa168** [comment]
Will attempt a merge in light of stable library and java8 updates, thanks!

**Pull request merged**